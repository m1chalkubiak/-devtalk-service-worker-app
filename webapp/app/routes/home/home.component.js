import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { FormattedMessage } from 'react-intl';

import messages from './home.messages';
import { Container, Title, TitleLogo } from './home.styles';

export class Home extends PureComponent {
  static propTypes = {
    language: PropTypes.string.isRequired,
    setLanguage: PropTypes.func.isRequired,
    match: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
  };

  render() {
    return (
      <Container>
        <Helmet title="Homepage" />

        <Title>
          <TitleLogo name="logo" />
          <FormattedMessage {...messages.welcome} />
        </Title>
      </Container>
    );
  }
}
