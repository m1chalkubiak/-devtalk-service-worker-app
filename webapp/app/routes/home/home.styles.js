import styled from 'styled-components';

import { SpriteImage } from '../../theme';

export const Container = styled.div`
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
 text-align: center;
`;

export const Title = styled.h1`
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
`;

export const TitleLogo = SpriteImage.extend`
  margin-bottom: 16px;
`;
