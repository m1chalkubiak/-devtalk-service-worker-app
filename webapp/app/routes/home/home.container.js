import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { hot } from 'react-hot-loader';
import { withRouter } from 'react-router-dom';
import { compose } from 'ramda';

import { Home } from './home.component';
import { LocalesActions } from '../../modules/locales/locales.redux';
import { selectLocalesLanguage } from '../../modules/locales/locales.selectors';


const mapStateToProps = createStructuredSelector({
  language: selectLocalesLanguage,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  setLanguage: LocalesActions.setLanguage,
}, dispatch);

export default compose(
  hot(module),
  connect(mapStateToProps, mapDispatchToProps),
  withRouter
)(Home);
