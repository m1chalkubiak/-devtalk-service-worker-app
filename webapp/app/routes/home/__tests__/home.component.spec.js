import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { Home } from '../home.component';


describe('Home: Component', () => {
  const defaultProps = {
    items: fromJS([{ name: 'name-1' }, { name: 'name-2' }, { name: 'name-3' }]),
    language: 'en',
    setLanguage: () => {},
    location: {},
    match: {},
    history: { push: () => {} },
  };

  const component = (props) => (
    <Home {...defaultProps} {...props} />
  );

  it('should render Home root', () => {
    const wrapper = shallow(component({}));
    global.expect(wrapper).toMatchSnapshot();
  });
});
