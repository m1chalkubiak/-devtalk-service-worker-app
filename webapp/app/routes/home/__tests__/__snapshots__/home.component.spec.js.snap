// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Component should render Home root 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Homepage"
  />
  <styled.h1>
    <styled.i
      name="logo"
    />
    <FormattedMessage
      defaultMessage="Hello World!"
      id="home.welcome"
      values={Object {}}
    />
  </styled.h1>
</styled.div>
`;
